tests/integration/ou/ouapi_test.go

```diff
@@ -0,0 +1,685 @@
+	updateRequest := UpdateOURequest{
+		Name:        "Non-existent OU Update",
+		Description: "Testing update of non-existent OU",
+	}
+
+	jsonData, err := json.Marshal(updateRequest)
+	suite.Require().NoError(err)
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	req, err := http.NewRequest("PUT", testServerURL+"/organization-units/non-existent-id", bytes.NewBuffer(jsonData))
+	suite.Require().NoError(err)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := client.Do(req)
+	suite.Require().NoError(err)
+	defer resp.Body.Close()
+
+	suite.Equal(http.StatusNotFound, resp.StatusCode)
+}
+
+func (suite *OUAPITestSuite) TestDeleteNonExistentOrganizationUnit() {
+	// Try to delete a non-existent OU
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	req, err := http.NewRequest("DELETE", testServerURL+"/organization-units/non-existent-id", nil)
+	suite.Require().NoError(err)
+
+	resp, err := client.Do(req)
+	suite.Require().NoError(err)
+	defer resp.Body.Close()
+
+	suite.Equal(http.StatusNotFound, resp.StatusCode)
+}
+
+func createOU(ts *OUAPITestSuite, ouRequest CreateOURequest) (string, error) {
+	jsonData, err := json.Marshal(ouRequest)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal OU request: %w", err)
+	}
+
+	req, err := http.NewRequest("POST", testServerURL+"/organization-units", bytes.NewBuffer(jsonData))
+	if err != nil {
+		return "", fmt.Errorf("failed to create request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("failed to send request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated {
+		bodyBytes, _ := io.ReadAll(resp.Body)
+		return "", fmt.Errorf("expected status 201, got %d: %s", resp.StatusCode, string(bodyBytes))
+	}
+
+	var createdOU OrganizationUnit
+	err = json.NewDecoder(resp.Body).Decode(&createdOU)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse response body: %w", err)
+	}
+
+	return createdOU.ID, nil
+}
+
+func deleteOU(ouID string) error {
+	req, err := http.NewRequest("DELETE", testServerURL+"/organization-units/"+ouID, nil)
+	if err != nil {
+		return err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusNoContent {
+		return fmt.Errorf("expected status 204, got %d", resp.StatusCode)
+	}
+	return nil
+}
+
+// Helper function to create string pointer
+func stringPtr(s string) *string {
+	return &s
+}
+
+func TestOUAPITestSuite(t *testing.T) {
```

Comment:
Shall we move this method right after the `type OUAPITestSuite struct` definition to maintain consistency and increase readability?

Reply:
Fixed with https://github.com/asgardeo/thunder/compare/4f9ca7c2768bc2448d03776f8028318bf6a25414..8604a0470ea625d3e6b6f7da269762e6aaf15252